File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    
    cnt=0
    def dfs(idx, sum):
        if idx==len(numbers):
            if(sum==target):
                nonlocal cnt
                cnt+=1
            return
        else:
            dfs(idx+1, sum+ numbers[idx])
            dfs(idx+1, sum- numbers[idx])
    dfs(0,0)
    return cnt

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    
    static int answer = 0;
    public int solution(int[] numbers, int target) {
        
        dfs(numbers, target, 0,0);
        return answer; 
        
    }
    
    static public void dfs(int[] numbers, int target, int depth, int sum){
        
        if(depth==numbers.length){ //DFS +-모두 탐색한 경우
            if(sum==target)
                answer++;
        }
        else{
            dfs(numbers, target, depth+1, sum + numbers[depth]);
            dfs(numbers, target, depth+1, sum - numbers[depth]);
        }
        
    }
           
}